src/app/core/model/data/constraint.ts

```diff
@@ -0,0 +1,155 @@
+/*
+ * Lumeer: Modern Data Definition and Processing Platform
+ *
+ * Copyright (C) since 2017 Answer Institute, s.r.o. and/or its affiliates.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {ColumnFunction} from './column-function';
+
+export enum ConstraintType {
+  Address = 'Address',
+  Boolean = 'Boolean',
+  Coordinates = 'Coordinates',
+  DateTime = 'DateTime',
+  Decimal = 'Decimal',
+  Email = 'Email',
```

Comment:
What about using `Contact` type which includes `Email`, `Phone` etc.. ?

Reply:
This PR tries to implement things described in _Use case: Constraints_ document. Feel free to suggest design changes there.